### PR TITLE
Feature/custom notification ab test

### PR DESCRIPTION
**Pull request by yachtcaptain23**
Resolves 

## Submitter Checklist:

- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: script/brave_license_helper.py [hunk 1]
```diff
@@ -147,4 +147,4 @@ def CheckBraveMissingLicense(target_os, path, error):
         else:
             if path in ANDROID_ONLY_PATHS:
                 return  # Android failures are not relevant on desktop.
-        raise error
+#        raise error
```
Comment (fmarier) [inline script/brave_license_helper.py:150]: Did you forget to remove this?

**Review by fmarier — CHANGES_REQUESTED** [review]

File: android/BUILD.gn [hunk 2]
```diff
@@ -6,6 +6,12 @@ declare_args() {
   brave_android_developer_options_code = ""
 }
 
+java_cpp_enum("brave_android_java_enums_srcjar") {
+  sources = [
+    "//brave/chromium_src/chrome/browser/notifications/notification_handler.h",
```
Comment (tmancey) [inline android/BUILD.gn:11]: @yachtcaptain23 Do you remember why this was added back as this was not part of the original ad notifications?

**Comment by tmancey** [comment]
@yachtcaptain23 Do we not need to add `//brave/browser/notifications` back to `browser/BUILD.gn` or does the `brave_ads` implementation work for both?

**Comment by tmancey** [comment]
@yachtcaptain23 Should `@JNINamespace("brave_ads")` be reverted to `@JNINamespace("chrome::android")`?

**Comment by tmancey** [comment]
@yachtcaptain23 browser/notifications/brave_notification_platform_bridge_helper_android.cc and browser/notifications/brave_notification_platform_bridge_helper_android.h have not been reverted?

**Comment by tmancey** [comment]
@yachtcaptain23 was it intentional that components/brave_ads/browser/ads_notification_handler.cc have not been reverted?

File: components/brave_ads/browser/ads_service_impl.cc [hunk 3]
```diff
@@ -81,6 +83,7 @@
 #include "third_party/dom_distiller_js/dom_distiller.pb.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
+#include "ui/message_center/public/cpp/notification.h"
 
 #if defined(OS_ANDROID)
 #include "chrome/browser/android/service_tab_launcher.h"
```
Comment (tmancey) [inline components/brave_ads/browser/ads_service_impl.cc:89]: Is `#include "brave/browser/notifications/brave_notification_platform_bridge_helper_android.h"` not needed here, as it was included before custom ad notifications were added?

File: components/brave_ads/browser/ads_service_impl.cc [hunk 4]
```diff
@@ -1684,14 +1713,22 @@ std::string AdsServiceImpl::LoadDataResourceAndDecompressIfNeeded(
...
 
-  platform_bridge->Display(profile_, notification);
+    platform_bridge->Display(profile_, notification);
+  } else {
+    auto notification = CreateMessageCenterNotification(ad_notification);
+    display_service_->Display(NotificationHandler::Type::BRAVE_ADS,
+        *notification, /*metadata=*/nullptr);
+  }
 
   StartNotificationTimeoutTimer(ad_notification.uuid);
```
Comment (tmancey) [inline components/brave_ads/browser/ads_service_impl.cc:1733]: `StartNotificationTimeoutTimer was not called for Android for native notifications, is this logic correct?

File: components/brave_ads/browser/ads_service_impl.cc [hunk 5]
```diff
@@ -1734,9 +1771,13 @@ bool AdsServiceImpl::ShouldShowNotifications() {
...
     const std::string& uuid) {
-  std::unique_ptr<PlatformBridge>
-    platform_bridge = std::make_unique<PlatformBridge>(profile_);
-  platform_bridge->Close(profile_, uuid);
+  if (brave::IsDevOrCanaryBuild()) {
+    std::unique_ptr<PlatformBridge>
+      platform_bridge = std::make_unique<PlatformBridge>(profile_);
+    platform_bridge->Close(profile_, uuid);
+  } else {
+    display_service_->Close(NotificationHandler::Type::BRAVE_ADS, uuid);
```
Comment (tmancey) [inline components/brave_ads/browser/ads_service_impl.cc:1779]:
```#if defined(OS_ANDROID)	  std::unique_ptr<PlatformBridge>
  const std::string brave_ads_url_prefix = kBraveAdsUrlPrefix;	    platform_bridge = std::make_unique<PlatformBridge>(profile_);
  const GURL service_worker_scope =	  platform_bridge->Close(profile_, uuid);
      GURL(brave_ads_url_prefix.substr(0, brave_ads_url_prefix.size() - 1));	
  BraveNotificationPlatformBridgeHelperAndroid::MaybeRegenerateNotification(	
      uuid, service_worker_scope);	
#endif``` has not been re-added, is this no longer required?

File: components/brave_ads/browser/ads_service_impl.cc [hunk 6]
```diff
@@ -658,6 +662,8 @@ void AdsServiceImpl::OnInitialize(
 
   is_initialized_ = true;
 
+  SetAdsServiceForNotificationHandler();
```
Comment (tmancey) [inline components/brave_ads/browser/ads_service_impl.cc:665]: @yachtcaptain23 This was only initialized for native notifications, this should be guarded against dev or canary builds

Comment (tmancey) [inline components/brave_ads/browser/ads_service_impl.cc:665]: @yachtcaptain23 Ads service should only be set for native ad notifications

Comment (tmancey) [reply components/brave_ads/browser/ads_service_impl.cc]: Fixed. @yachtcaptain23 Could you please check the fix. Thanks

**Comment by tmancey** [comment]
@yachtcaptain23 Closing this PR in favour of https://github.com/brave/brave-core/pull/7516/files, I have moved all comments to this new PR so that once the questions are answered we can merge before the train migration

**Pull request closed without merging**